src/api/middleware/validators/validatorUtils.js

```diff
@@ -61,10 +61,36 @@ const isObjectId = (id) => {
  */
 const concurrentOffersNotExceeded = (OfferModel) => async (owner, publishDate, publishEndDate) => {
     // We need to pass the offer model in case we're inside an Offer instance
-    const concurrentOffers = await (new CompanyService())
+    const offersInTimePeriod = await (new CompanyService())
```

Comment:
Correct me if I'm wrong but I don't think that's necessary. I think that, since all those offers collide with the candidate offer, the number of concurrent offers will always be what I'm currently calculating plus one (hence the `<` instead of the `<=`.
Is there any counterexample that I'm not seeing?

Reply:
Not necessarily, you can have 5 max concurrent offers at a given time, and add a new one that does not make it 6, that's why you need to include it in the algorithm:

```
-----XXX  <- New offer
XXXXXX
XXXXX
XXXXX
XXXXX
XXXXX
```

For the first check it's ok, since you only check for total intersections, but for the actual algorithm, it neds to take into account the new offer, and for that you might want to change the first section as well, to take the new offer into account

Reply:
I can include it to remove all doubts but, in that examples, the date arrays wouldn't include the bottom 4 offers, since we're only fetching offers which collide with the new offer from the database (not all of the company's offers)

Reply:
Hmm, I see, you're also using the < only in the last check, so it should work. I could not find a counter example